File: android/app/src/main/java/com/brekeke/phonedev/utils/AccountUtils.java
package com.brekeke.phonedev.utils;

import org.json.JSONArray;
import org.json.JSONObject;

public class AccountUtils {
  private static boolean compareFalsishField(String v1, String v2) {
    return isNullOrEmpty(v1) || isNullOrEmpty(v2) || v1.equals(v2);
  }

  private static boolean isNullOrEmpty(String s) {
    return s == null || s.isEmpty();
  }

  public static JSONObject findAccountPartial(JSONArray profilesArray, String username, String tenant, String host, String port) {
    for (int i = 0; i < profilesArray.length(); i++) {
      JSONObject acc = profilesArray.optJSONObject(i);
      if (acc == null) continue;

      String accUsername = acc.optString("pbxUsername", "");
      if (isNullOrEmpty(accUsername)) continue;

      if (accUsername.equals(username)
              && compareFalsishField(acc.optString("pbxTenant", ""), tenant)
              && compareFalsishField(acc.optString("pbxHostname", ""), host)
              && compareFalsishField(acc.optString("pbxPort", ""), port)) {
        return acc;
      }
    }
    return null;
  }
}

File: ios/BrekekePhone/Utils/AccountUtils.swift
import Foundation

public class AccountUtils {
  private static func compareFalsishField(v1: String, v2: String) -> Bool {
    return v1.isEmpty || v2.isEmpty || v1 == v2
  }

  static func find(username: String, tenant: String, host: String,
                   port: String) -> Profile? {
    if let a = Storage.account() {
      for p in a {
        if p.pbxUsername.isEmpty {
          continue
        }
        if p.pbxUsername == username,
           compareFalsishField(v1: p.pbxTenant ?? "", v2: tenant),
           compareFalsishField(v1: p.pbxHostname ?? "", v2: host),
           compareFalsishField(v1: p.pbxPort ?? "", v2: port) {
          return p
        }
      }
    }
    return nil
  }

  static func find(m: [AnyHashable: Any]) -> Profile? {
    let u = PN.username(m) ?? ""
    let t = PN.tenant(m) ?? ""
    let h = PN.host(m) ?? ""
    let p = PN.port(m) ?? ""
    return find(username: u, tenant: t, host: h, port: p)
  }
}
